public/app/features/plugins/admin/helpers.ts

```diff
@@ -119,7 +120,7 @@ export function mapLocalToCatalog(plugin: LocalPlugin, error?: PluginError): Cat
     installedVersion: version,
     hasUpdate,
     isInstalled: true,
-    isDisabled: !!error,
+    isDisabled: !enabled || !!error,
```

Comment:
I was expecting this to incorrectly show disabled badge for non-enabled app plugins, but it doesn't. Tested with the Zabbix plugin. What am I missing here?

Reply:
there is a mapLocalToCatalog and a mapRemoteToCatalog, is Zappix remote and thus doesn't hit this change?

Reply:
Both local and remote I guess

Reply:
Turns out it is not very easy. If a plugin is disabled, the UI renders the disabled plugin page and you lose the ability to Enable it. I would recommend we just keep this change as-is and open a follow up task for other plugin types.

Reply:
Looked at this a bit more and think your solution only works when there are a local plugin, but not a emote one. So whenever a secret storage plugin is published on grafana.com your suggested solution would not work any longer. Might need to introduce a new property instead of using the enabled one since that is mainly used for app plugins which can be disabled/enabled per organization.

I'm not sure the suggested solution of showing this in the plugin catalog make sense, but I understand why you decided to go this path short term. If you have configured secret storage plugin and there was a failure to use the plugin then maybe Grafana shouldn't even start? How is this suppose to work, are you manually going into the plugins catalog to verify it works or not? Given you have an existing Grafana instance with secrets stored and restarting that instance with failure to load the secret storage plugin you would be in a weird situation. Tricky.

We do have similar problem with the image renderer plugin, but for that we show in the UI whenever it's not available, e.g. https://github.com/grafana/grafana/pull/23856. This is a config property made available client side via https://github.com/grafana/grafana/blob/53d03aec785d8d9282697218751a6d323bd76af9/pkg/api/frontendsettings.go#L152. A possible alternative to adding a new plugin property in response could be to add something similar and check this config in the plugins catalog while loading plugins. Doesn't really make a huge difference I guess.

Reply:
Ah, I see what you're mean. 

The situation you described is tricky because the danger of having the plugin installed but not enabled is dependent on whether migration has happened yet. I think that is a conversation we will have to have as a team and determine the best course of action.

I could add a new special property to the entire plugin object to handle the secret manager case, or perhaps it would be better to revert the server side changes and just create an error when a secrets manager is installed but disabled.

Reply:
Changed logic for showing the disabled badge. Is now highly specific to the secrets plugin type. Not ideal, but I know we will be reworking this eventually.